liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/applications/LooseWarApplication.java

```diff
@@ -12,17 +23,244 @@
+        return isFilteringDeploymentDescriptors(project);
+    }
+    
+    public static boolean isUsingOverlays(MavenProject project) {
+    	boolean overlaysEnabled = false;
+    	
+    	// Get overlay dependencies
+    	List<Dependency> overlayDependencies = getOverlayDependencies(project);
+    	
+    	// Get overlays configured in the Maven WAR plugin
+    	List<Xpp3Dom> overlayConfigurations = getOverlayConfigurations(project);
+    	
+    	if (!overlayDependencies.isEmpty() || !overlayConfigurations.isEmpty()) {
+    		overlaysEnabled = true;
+    	}
+    	
+    	return overlaysEnabled;
+    }
+    
+    /**
+     * Get overlay configuration values from the Maven WAR plugin
+     * @param proj the Maven project
+     * @return ALLOWS DUPS
+     * @return a List of war plugin overlay elements
+     */
+    private static List<Xpp3Dom> getOverlayConfigurations(MavenProject project) {
+        List<Xpp3Dom> retVal = new ArrayList<Xpp3Dom>();
+        Xpp3Dom dom = project.getGoalConfiguration("org.apache.maven.plugins", "maven-war-plugin", null, null);
+        if (dom != null) {
+            Xpp3Dom overlays = dom.getChild("overlays");
+            if (overlays != null) {
+                Xpp3Dom overlayList[] = overlays.getChildren("overlay");
+                if (overlayList != null) {
+                    for (int i = 0; i < overlayList.length; i++) {
+                        retVal.add(overlayList[i]);
+                    }
+                }
+            }
+        }
+        return retVal;
     }
     
-    public void addSourceDir(MavenProject proj) throws Exception {
-        File sourceDir = new File(proj.getBasedir().getAbsolutePath(), "src/main/webapp");
-        String path = MavenProjectUtil.getPluginConfiguration(proj, "org.apache.maven.plugins", "maven-war-plugin", "warSourceDirectory");
-        if (path != null) {
-            sourceDir = new File(proj.getBasedir().getAbsolutePath(), path);
-        } 
-        config.addDir(sourceDir, "/");
+    /**
+     * Get overlay dependencies
+     * @param proj the Maven project
+     * @return ALLOWS DUPS
+     * @return a List of war plugin overlay dependencies
+     */
+    private static List<Dependency> getOverlayDependencies(MavenProject project) {
+    	List<Dependency> overlayDependencies = new ArrayList<Dependency>();
+    	
+    	List<Dependency> deps = project.getDependencies();
+    	for (Dependency dep : deps) {
+    		if (dep.getType().equals("war")) {
+    			overlayDependencies.add(dep);
+    		}
+    	}
+    	
+    	return overlayDependencies;
+    }
+
+
+    /**
+     * Get resource configuration values that have "directory" children elements from the Maven WAR plugin
+     * @param proj the Maven project
+     * @return ALLOWS DUPS
+     * @return a List of war plugin resource elements that contain a "directory" child element
+     */
+    public static List<Xpp3Dom> getWebResourcesConfigurations(MavenProject project) {
```

Comment:
similar comment.. we could make this a non-static.   Maybe the totality of the code suggests static.  And not a huge deal if there's no NPE.